src/ray/core_worker/reference_count.h

```diff
@@ -49,9 +51,17 @@ class ReferenceCounter {
   /// Add references for the provided object IDs that correspond to them being
   /// dependencies to a submitted task.
   ///
-  /// \param[in] object_ids The object IDs to add references for.
+  /// \param[in] argument_ids_to_add The arguments of the task to add
+  /// references for.
+  /// \param[in] pin_lineage Whether to increment the arguments' lineage ref
+  /// count. If true, then the Reference entry for each new argument will not
+  /// be deleted until the arguments' lineage ref is released.
+  /// \param[out] argument_ids_to_remove The arguments of the task to remove
+  /// references for.
+  /// \param[out] deleted Any objects that are newly out of scope after this
+  /// function call.
   void UpdateSubmittedTaskReferences(
-      const std::vector<ObjectID> &argument_ids_to_add,
+      const std::vector<ObjectID> &argument_ids_to_add, bool pin_lineage,
```

Comment:
Instead of toggling whether to increment or decrement the lineage count, how about we always do the accounting, but use the flag to decide whether to evict even if lineage count is > 0?

Reply:
Hmm good idea, but we'll likely have to add it back once it's possible to resubmit tasks that have finished before.